marlowe-dashboard-client/src/Capability/Websocket.purs

```diff
@@ -0,0 +1,36 @@
+module Capability.Websocket
+  ( class MonadWebsocket
+  , subscribeToWallet
+  , unsubscribeFromWallet
+  , subscribeToContract
+  , unsubscribeFromContract
+  ) where
+
+import Prelude
+import AppM (AppM)
+import Bridge (toBack)
+import Data.Either (Either(..))
+import Halogen (HalogenM, raise)
+import MainFrame.Types (Msg(..))
+import Plutus.PAB.Webserver.Types (CombinedWSStreamToServer(..))
+import WalletData.Types (Wallet)
+import Types (ContractInstanceId)
+
+class
+  Monad m <= MonadWebsocket m where
```

Comment:
These seems to belong to the Wallet and Contract API

Reply:
Hmm... :thinking: The problem with that is that the `HalogenM` instance here needs to be more specific (it needs to have the right `Msg` type for sending websocket messages). Also I was thinking of the other modules as just for API calls - sending messages through the websocket is a different matter.